naga/src/back/glsl/mod.rs

```diff
@@ -4774,7 +4774,7 @@ const fn glsl_sampling(sampling: crate::Sampling) -> Option<&'static str> {
     use crate::Sampling as S;
 
     match sampling {
-        S::Center => None,
+        S::Center | S::First | S::Either => None,
```

Comment:
We should error if we see `first` in the GLSL backend.

Reply:
I'm thinking of using a new error variant with this message:

```
`@interpolate(flat, first)` is unsupported in the GLSL backend
```

Eventually, we can add a "reason" field to the error message, which we could use to differentiate rejection between this case and using WebGPU compat.

Reply:
...oh, but this is a backend context, where we wouldn't be checking for `compat`. Nevermind, then, just gonna add something to `naga::back::glsl::Error`, then. 😅

Reply:
Yeah, I was thinking of adding a new variant to GLSL's error type. GLSL will always be compat so that's fine.

Reply:
With my latest force push to 0591c9a5, I've added some test coverage to ensure that we're getting this:

* A new `validation::no_flat_first_in_glsl` test that ensures we emit a new `naga::back::glsl::Error::FirstSamplingNotSupported` variant, as we discussed above.
* Removal of the GLSL backend from `tests/in/interpolate.wgsl` snapshots, since `@interpolate(flat, first)` isn't testable in GLSL.
* A new `tests/in/interpolate_compat.wgsl` snapshot test that's like `interpolate.wgsl`, but omits the `@interpolate(flat, first)` test case and _does_ still snapshot GLSL output.